src/pages/strategies/forms/strategiesForm.js

```diff
@@ -0,0 +1,163 @@
+      groupId: yup.string().required('This field is required'),
+      type: yup.string().required('This field is required')
+    }),
+    onSubmit: values => {
+      postGroups(values)
+    }
+  })
+
+  const prevValuesRef = useRef()
+
+  useEffect(() => {
+    if (prevValuesRef.current && formikValuesHaveChanged(prevValuesRef.current, strategiesFormik.values)) {
+      onUpdateFormik(strategiesFormik)
+    }
+    prevValuesRef.current = strategiesFormik.values
+  }, [strategiesFormik.values, onUpdateFormik])
+
+  const formikValuesHaveChanged = (prevValues, newValues) => {
+    return JSON.stringify(prevValues) !== JSON.stringify(newValues)
+  }
+
+  const postGroups = async obj => {
+    const isNewRecord = !obj?.recordId
+    try {
+      const res = await postRequest({
+        extension: DocumentReleaseRepository.Strategy.set,
+        record: JSON.stringify(obj)
+      })
+
+      const message = isNewRecord ? 'Record Added Successfully' : 'Record Edited Successfully'
+      toast.success(message)
+
+      if (isNewRecord) {
+        setStore(prevStore => ({
+          ...prevStore,
+          recordId: res.recordId
+        }))
+      }
+      setEditMode(true)
+      invalidate()
+    } catch (error) {
+      toast.error('An error occurred')
+    }
+  }
+
+  useEffect(() => {
+    if (recordId) {
+      getGroupId(recordId)
+    }
+  }, [recordId, getRequest])
+
+  const getGroupId = recordId => {
+    const defaultParams = `_recordId=${recordId}`
+    getRequest({
+      extension: DocumentReleaseRepository.Strategy.get,
+      parameters: defaultParams
+    })
+      .then(res => {
+        setInitialData(res.record)
+        setEditMode(true)
+      })
+      .catch(error => {
+        console.error('Error fetching group ID:', error)
+      })
+  }
+
+  return (
+    <FormShell form={strategiesFormik} resourceId={ResourceIds.Strategies} maxAccess={maxAccess} editMode={editMode}>
+      <Grid container spacing={4}>
+        <Grid item xs={12}>
+          <CustomTextField
+            name='name'
+            label={labels.name}
+            value={strategiesFormik.values.name}
+            required
+            maxLength='50'
+            maxAccess={maxAccess}
+            onChange={strategiesFormik.handleChange}
+            onClear={() => strategiesFormik.setFieldValue('name', '')}
+            error={strategiesFormik.touched.name && Boolean(strategiesFormik.errors.name)}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <ResourceComboBox
+            endpointId={DocumentReleaseRepository.DRGroup.qry}
+            parameters='_startAt=0&_pageSize=100'
+            name='groupId'
+            label={labels.groupStrat}
+            valueField='recordId'
+            displayField='name'
+            values={strategiesFormik.values}
+            required
+            readOnly={editMode}
+            maxAccess={maxAccess}
+            onClear={() => formik.setFieldValue('codeId', '')}
```

Comment:
do we need this or the resourceCombobox does the job ? 
same for the onChange event.

Reply:
no we dont need the onClear  i remove it 
but i need the onChange